### PR TITLE
[1/2] DXCDT-426: Add new `auth0_connection_clients` resource

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

In this PR we are adding a brand new `auth0_connection_clients` resource that managed the relationship between one connection and its respective enabled clients. 

> **Warning**
> This new resource should not be used together with the `auth0_connection_client` resource, as they both operate on the same property of the connection, but one does so in bulk and the other is append only. 

> **Note**
> In this PR we are not adding the logic to detect if both the 1:1 and 1:many resources for the connection are used within the same config. That will be a separate follow up through DXCDT-446. However that will only raise issues within the same config. It's still advised to not use both resources to manage the enabled clients of the same connection even if they are in separate terraform configurations.

### 📚 References

- #549 
- #546 

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: docs/resources/connection_client.md [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
-  With this resource, you can manage enabled clients on a connection.
+  With this resource, you can enable a single client on a connection.
 ---
 
 # Resource: auth0_connection_client
 
-With this resource, you can manage enabled clients on a connection.
+With this resource, you can enable a single client on a connection.
+
+!> To prevent issues, avoid using this resource together with the `auth0_connection_clients` resource.
```
Comment (sergiught) [inline docs/resources/connection_client.md:11]: https://developer.hashicorp.com/terraform/registry/providers/docs#callouts Red warning.

File: docs/resources/connection_client.md [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
 
 ## Example Usage
 
 ```terraform
 resource "auth0_connection" "my_conn" {
   name     = "My-Auth0-Connection"
   strategy = "auth0"
-  # Avoid using the enabled_clients = [...],
```
Comment (sergiught) [inline docs/resources/connection_client.md:17]: This is from the time we allowed managin the enabled_clients on the connection resource. Should not be here any more.

File: internal/auth0/connection/resource_clients.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+		Schema: map[string]*schema.Schema{
+			"connection_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of the connection on which to enable the client.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
```
Comment (sergiught) [inline internal/auth0/connection/resource_clients.go:28]: The name and strategy are readonly.

File: internal/auth0/connection/resource_clients_test.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+`
+
+const testAccConnectionClientsWithNoEnabledClients = givenASingleConnection + `
+resource "auth0_connection_clients" "my_conn_client_assoc" {
+	depends_on = [ auth0_connection.my_conn ]
+
+	connection_id = auth0_connection.my_conn.id
+	enabled_clients = []
+}
+`
```
Comment (willvedd) [inline internal/auth0/connection/resource_clients_test.go:106]: Idea: test the deletion of this resource by using a connection data source to check how many enabled clients there are.

**Review by willvedd — APPROVED** [review]
Overall solid and sensible contribution. A few ideas but I consider them non-blocking and could be implemented later.

File: internal/auth0/connection/resource_clients.go [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+			data.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	data.SetId(connection.GetID())
+
+	if len(connection.GetEnabledClients()) != 0 {
```
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/auth0/connection/resource_clients.go:77]: Good idea, I'll improve the logic for this check in a follow up PR and I'll post the link here, so we can review before merging this one too.

File: internal/auth0/connection/resource_clients.go [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+		data.SetId("")
+
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Connection with non empty enabled clients",
+				Detail: "The connection already has enabled clients attached to it. " +
```
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/auth0/connection/resource_clients.go:84]: I was trying to do that, but I couldn't find any way to get the name of the resource. 

Given this resource 

```terraform
resource auth0_connection_clients some_custom_name {
...
}
```

The import command would be: `terraform import auth0_connection_clients.some_custom_name <conn_id>`. 

We could just print it anyway like that tho `terraform import auth0_connection_clients.<your-resources-name> conn_123123` wdyt?

File: internal/auth0/connection/resource_clients.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/auth0/connection/resource_clients.go:77]: Followed up in https://github.com/auth0/terraform-provider-auth0/pull/570.

File: internal/auth0/connection/resource_clients.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/auth0/connection/resource_clients.go:84]: Followed up in https://github.com/auth0/terraform-provider-auth0/pull/570.

File: internal/auth0/connection/resource_clients_test.go (hunk 4 above)
Comment (sergiught) [reply internal/auth0/connection/resource_clients_test.go:106]: Followed up in https://github.com/auth0/terraform-provider-auth0/pull/570.

**Pull request merged**